### PR TITLE
Add Xcode Instruments to the "Using C++ profilers with Godot" page.

### PR DESCRIPTION
**Pull request by bruvzg**

### CODE REVIEW

File: development/cpp/using_cpp_profilers.rst [hunk 1]
```diff
@@ -141,3 +142,36 @@ HotSpot
...
+
+Xcode Instruments
+^^^^^^^^^^^^^^^^^
+
+- Open Xcode. Select **Open Developer Tool** - **Instruments** from the **Xcode** app menu:
+- Double-click on **Time Profiler** in the **Instruments** window:
+
+.. image:: img/cpp_profiler_xcode_menu.png
+
+- In the Time Profiler window, click on the **Target** menu select **Choose target...**
```
Comment (mhilbrunner) [inline development/cpp/using_cpp_profilers.rst:154]:
```suggestion
- In the Time Profiler window, click on the **Target** menu, select **Choose target...**
```

File: development/cpp/using_cpp_profilers.rst [hunk 2]
```diff
@@ -141,3 +142,36 @@ HotSpot
...
+  and specify the path to the Godot binary, command line arguments and environment variables
+  in the next window.
+
+.. image:: img/cpp_profiler_time_profiler.png
+
+- You can also attach Time Profiler to a running process by selecting it from the **Target**
```
Comment (mhilbrunner) [inline development/cpp/using_cpp_profilers.rst:160]:
```suggestion
- You can also attach the Time Profiler to a running process by selecting it from the **Target**
```

File: development/cpp/using_cpp_profilers.rst [hunk 3]
```diff
@@ -141,3 +142,36 @@ HotSpot
...
+  menu.
+
+- Click the **Start an immediate mode recording** button to start profiling.
+
+.. image:: img/cpp_profiler_time_profiler_record.png
+
+- Perform the actions you wish to profile in the editor or project. When you're done,
+  click **Stop** button.
```
Comment (mhilbrunner) [inline development/cpp/using_cpp_profilers.rst:168]:
```suggestion
  click the **Stop** button.
```

File: development/cpp/using_cpp_profilers.rst [hunk 4]
```diff
@@ -141,3 +142,36 @@ HotSpot
...
+
+- Wait for the results to appear.
+- At the bottom of the window you will see a call tree for all CPU threads started, and
+  **Heaviest Stack Trace** overview.
```
Comment (mhilbrunner) [inline development/cpp/using_cpp_profilers.rst:172]:
```suggestion
  the **Heaviest Stack Trace** overview.
```

File: development/cpp/using_cpp_profilers.rst [hunk 5]
```diff
@@ -141,3 +142,36 @@ HotSpot
...
+- Select **Hide system libraries** in the **Call Tree** menu (at the bottom of window) to
+  remove external modules.
+- You can use timeline at the top of the window to display details for the specific time period.
```
Comment (mhilbrunner) [inline development/cpp/using_cpp_profilers.rst:175]:
```suggestion
- You can use the timeline at the top of the window to display details for the specific time period.
```

**Comment by mhilbrunner** [comment]
Thank you!

**Comment by Calinou** [comment]
Cherry-picked to the `3.3` and `stable` branches.

**Pull request merged**